fix: Improve mobile responsiveness and code block button visibility

mohamedhadri opened the pull request:
## Description

This PR fixes mobile responsiveness issues by implementing comprehensive CSS improvements and configuration updates to enhance the documentation site's usability on mobile devices. The changes ensure proper display, prevent horizontal scrolling, and improve accessibility for users on phones and tablets.

**Key Improvements:**

**Mobile Responsiveness and Usability:**
- Made code block copy buttons and button groups always visible on mobile devices (no hover required), enabling users to easily copy code and prevent horizontal scrolling
- Added comprehensive mobile-specific CSS rules (`@media max-width: 996px`) for:
  - Navigation menus with better touch targets (minimum 44px for accessibility)
  - Responsive images that scale properly without causing overflow
  - Horizontally scrollable tables on mobile
  - Optimized typography with adjusted heading sizes and line heights
  - Improved sidebar layout and spacing
  - Better button and interactive element sizing
  - Enhanced footer layout for small screens
  - Prevention of horizontal scroll with proper overflow handling

**Configuration Enhancements:**
- Updated `docusaurus.config.js` to include a `viewport` meta tag ensuring proper scaling and rendering on mobile devices

**Files Modified:**
- `src/css/custom.css` - Added 250+ lines of mobile-responsive CSS
- `docusaurus.config.js` - Added viewport meta tag configuration

## Related Issues

Closes #269 

## Mobile & Desktop Screenshots/Recordings

<!-- Will add screenshots showing before/after mobile view in comments -->

**Before:** Code block buttons invisible on mobile, horizontal scrolling required
<img width="535" height="881" alt="image" src="https://github.com/user-attachments/assets/6158574f-16c7-45c3-999e-8dffd97f4f49" />

**After:** Buttons always visible, no horizontal scroll, properly scaled images and layout
<img width="507" height="816" alt="image" src="https://github.com/user-attachments/assets/f6219760-51cb-41f1-97c6-75b7fb64c273" />


## Steps to QA

1. Open the documentation site on your browser
2. Switch to mobile view (DevTools > Toggle device toolbar) or test on actual mobile device
3. Navigate to any page with code blocks (e.g., "How to Contribute to Open Source")
4. Verify that copy/wrap buttons are visible without hovering
5. Check that code blocks don't cause horizontal scrolling
6. Test image scaling - all images should fit within viewport
7. Open the navigation menu - verify touch targets are adequate (easy to tap)
8. Test table scrolling - tables should scroll horizontally if needed
9. Verify no horizontal scroll on any page
10. Check that all interactive elements are easily tappable

**Test on multiple screen sizes:**
- Mobile (320px - 480px)
- Tablet (768px - 996px)
- Desktop (997px+)

## What GIF best describes this PR or how it makes you feel?

![mobile-friendly](https://media.giphy.com/media/3o7abKhOpu0NwenH3O/giphy.gif)

*Because now mobile users can actually read the docs without doing the horizontal scroll dance! 📱✨*

mohamedhadri committed:
fix: Improve mobile responsiveness and code block button visibility

mohamedhadri committed:
refactor: Remove article padding from dark theme styles

mohamedhadri committed:
style: Enhance mobile layout by removing padding from containers and articles

adiati98 reviewed (changes requested):
Hey @mohamedhadri,

Thanks for the PR!

I want to address some things here:

1. I see that the code block buttons are now visible on the small screen. Thanks!
2. Do we need all the `!important` here? It seems some of them are causing new issues, such as:

   - The logo at the footer become much bigger
   - The previous and next page navigations are cut off

       **Before:**

       <img width="1297" height="1333" alt="Image" src="https://github.com/user-attachments/assets/441bddff-c778-4b75-b80a-6ae4ec967965" />

       **After:**
      
       <img width="1354" height="1349" alt="Image" src="https://github.com/user-attachments/assets/aae68d1a-d76f-4104-851f-f59b51acd297" />

3. The code block overflow is not addressed yet. You might want to visit the [overflow](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow) CSS property to achieve this.

Can you please address these? Thank you.

BekahHW committed:
Merge branch 'main' into fix/improve-mobile-responsiveness

adiati98 committed:
Merge branch 'main' into fix/improve-mobile-responsiveness

mohamedhadri committed:
feat: add mobile word wrap functionality and improve image responsiveness

mohamedhadri commented:
hey @adiati98, 
I pushed a new commit now and here is the summary of it:

- Fixed Footer Logo Size on Mobile
The footer logo was oversized on mobile devices. It's now properly constrained to a maximum width of 72px for better visual balance:
<img width="394" height="489" alt="image" src="https://github.com/user-attachments/assets/1b71a43c-f750-4d68-b8f0-639ec5075950" />

- Restored Navigation to Original Design
The navigation menu has been restored to its original appearance while maintaining mobile responsiveness:

<img width="409" height="275" alt="image" src="https://github.com/user-attachments/assets/bf32d09e-1606-4749-9515-1d1036b49afe" />


-Enhanced Code Block UX on Mobile - Word Wrap Enabled by Default
Implemented automatic word wrap activation for code blocks on mobile devices. This significantly improves readability by eliminating horizontal scrolling. The word wrap button is now toggled on by default when viewing on mobile screens (≤ 996px width):

<img width="397" height="582" alt="image" src="https://github.com/user-attachments/assets/6e017c7b-f373-4ef1-b204-bb81ce31f59b" />

adiati98 committed:
Merge branch 'main' into fix/improve-mobile-responsiveness

adiati98 reviewed (approved):
Hey @mohamedhadri,

Apologize for the super late response!

This works as expected. Thank you! 🚀 

![Now, that is how you do it](https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExdnY2MjJtbTg1MzNudWY5ZTc0bDhrbXduOXdjN2czMDNranBucmtzYiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/KbGsQHbP3WO1M6HUpY/giphy.gif)
